modules/express/src/clientRoutes.ts

```diff
@@ -630,7 +639,7 @@ export async function handleV2OFCSignPayload(req: express.Request): Promise<{ pa
 
   const walletPassphrase = bodyWalletPassphrase || getWalletPwFromEnv(wallet.id());
   const tradingAccount = wallet.toTradingAccount();
-  const stringifiedPayload = JSON.stringify(req.body.payload);
+  const stringifiedPayload = isJsonString(req.body.payload) ? req.body.payload : JSON.stringify(req.body.payload);
```

Comment:
As a follow up, please consider migrating this api to an api-ts typed route! This way you can define your input codec and have your input validation done automatically!

Reply:
io-ts and io-ts-types already have codecs for json and json strings.

Reply:
Yeah, that was my original thoughts is that the payload strings could be decoded using same codec as the payload in the request bodies (super safe and well typed).